projects/storefrontapp-e2e-cypress/cypress/helpers/cart-import-export.ts

```diff
@@ -209,19 +209,23 @@ export function verifyImportedData(config: ImportConfig, cart) {
  * @param productCode identifies the unique product to add.
  */
 export function addProductToCart(productCode: string = cart.products[1].code) {
-  cy.intercept('GET', `**/users/*/carts/*?fields=**`).as('refresh_cart');
+  cy.intercept('GET', `**/products/${productCode}**`).as('productPage');
```

Comment:
any time there's a cy.visit page, you should use 

either the function for `waitForProductPage` for product details page or the generic `waitForPage` function.

Both functions can be pulled from the checkout helper in `projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-flow.ts`

Reply:
For this case, 

do something like

```
const productPage = waitForProductPage(productCode, 'getProductPage');
cy.visit(`/product/${productCode}`);
cy.wait(`@${productPage}`).its('response.statusCode').should('eq', 200);
```

P.S notice there's an extra assertion to make sure we do get a response back, where we check the status code because without it, it will just wait for the request to fire without having it resolved and get a proper response.

`.its('response.statusCode').should('eq', 200);`